### DISCUSSION TITLE
Suggestions for Dynamically Enabling/Disabling Menu Items

### DISCUSSION THREAD

**Discussion by joesken** [SnowdogApps/magento2-menu#317] [Q&A] [answered]
Hi,

I was looking for any suggestions on an approach to the following use case.

We have a category 'Special Offers' which only has products intermittently. We would like to dynamically add and remove the sub-menu item for it depending on if there are any products in the category.

Wondering if anyone has encountered this before and/or if you had any suggestions on an approach to this.

Thank you.

Joseph

**Comment by andrzejborkowski** [comment, marked as answer]
Hi @joesken

I created an internal ticket DEV-105318 to check the possibility of adding such a feature.
We are also working on Scheduler functionality for our menu module similar to https://experienceleague.adobe.com/en/docs/commerce-admin/catalog/products/product-scheduled-changes so this could be helpful in your case. It doesn't cover "empty" category condition however you'll able to schedule in time showing particular node in your menu.

Regards,
Andrzej

**Reply by joesken** [reply]
Hi @andrzejborkowski - thanks a million for coming back to me on this and so quickly. 

Just to say however in no way was I expecting you to add such a feature for me.  It genuinely was more of an open question looking for any suggestions on how it might be approached rather than creating further work for you.  Appreciated either way.

Thanks also for the notification on the Scheduler functionality. Good to know. 

Regards,
Joseph

**Reply by andrzejborkowski** [reply]
I think it will be useful feature for our menu module, so I put it on our dev's list already but I can't promise any deadlines for this. Once it's ready I'll update the issue so you can pull the code to your project. Have a great day!
